python/cuml/cuml/tests/test_pca.py

```diff
@@ -304,6 +304,22 @@ def test_sparse_pca_inputs(nrows, ncols, whiten, return_sparse, cupy_input):
         assert array_equal(i_sparse, X.todense(), 1e-1, with_sign=True)
 
 
+@pytest.mark.parametrize(
+    "n_samples, n_features",
+    [
+        pytest.param(9, 20, id="n_samples <= n_components"),
+        pytest.param(20, 10, id="n_features <= n_components"),
+    ],
+)
+def test_noise_variance_zero(n_samples, n_features):
```

Comment:
Can a test other than zero be added? To check the correctness of the computation